nomad/state/state_store.go

```diff
@@ -359,6 +370,22 @@ func (s *StateStore) UpsertPlanResults(msgType structs.MessageType, index uint64
 	txn := s.db.WriteTxnMsgT(msgType, index)
 	defer txn.Abort()
 
+	// Mark nodes as ineligible.
+	now := time.Now().Unix()
```

Comment:
Since UpsertPlanResults is called by the FSM we cannot check wallclock time here as it will differ on each server creating skew in their statestores. Since it's only informational there shouldn't be a negative functional impact, but it's still preferable to keep our FSMs deterministic on every server.

Reply:
Fixed in https://github.com/hashicorp/nomad/pull/13421/commits/fb2e761cda83d847960df140a743537db0827d2d